### PR TITLE
Cu 85ztre9m5 game logic ronisha shrestha

### PR DESCRIPTION
**Pull request by Ronishaaa**

### CODE REVIEW

**Comment by rubek-joshi** [comment]
Task linked: [CU-85ztre9m5 Game Logic](https://app.clickup.com/t/85ztre9m5)

File: src/context/GameContext.tsx [hunk 1]
```diff
@@ -79,11 +64,25 @@ export const GameContextProvider = (props: { children: React.ReactNode }) => {
...
 
+  const shuffle = (array: IQuiz[]) => {
+    return array
+      .map((a) => ({ sort: Math.random(), value: a }))
+      .sort((a, b) => a.sort - b.sort)
+      .map((a) => a.value);
+  };
+
+  useEffect(() => {
+    const data = mockData;
```
Comment (rubek-joshi) [inline src/context/GameContext.tsx:75]: This assignment is unnecessary

Comment (rubek-joshi) [reply src/context/GameContext.tsx:75]: Also change `mockData `to simply `data`

File: src/hooks/useClipboard.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  const handleCopyResult = (value: boolean) => {
+    clearTimeout(copyTimeout);
+    setCopyTimeout(setTimeout(() => setCopied(false), timeout));
+    setCopied(value);
+  };
+
+  const copy = (valueToCopy: string) => {
+    if ("clipboard" in navigator) {
+      console.log("here");
```
Comment (rubek-joshi) [inline src/hooks/useClipboard.tsx:19]: Remove console.logs

File: src/hooks/useClipboard.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      navigator.clipboard
+        .writeText(valueToCopy)
+        .then(() => {
+          console.log("tse");
```
Comment (rubek-joshi) [inline src/hooks/useClipboard.tsx:23]: Remove console.logs. Check everywhere.

File: src/screens/GameOver/components/Share.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import { useClipboard } from "hooks";
+import styles from "./Share.module.scss";
```
Comment (rubek-joshi) [inline src/screens/GameOver/components/Share.tsx:2]: Organize imports, relative imports should always be at last.

File: src/screens/GameOver/components/Share.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  BiLogoTwitter,
+} from "react-icons/bi";
+
+const shareData = {
+  title: "Cancer Awareness",
+  url: import.meta.env.VITE_BASE_URL,
+};
+
+export const Share = (props: { toggleShare: () => void }) => {
+  const { toggleShare } = props;
```
Comment (rubek-joshi) [inline src/screens/GameOver/components/Share.tsx:24]: De-structure above not here

File: src/screens/GameOver/components/Share.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    }
+  }, []);
+
+  const handleButtonClick = () => {
+    if (showShare) {
+      navigator.share(shareData).catch((error) => {
+        console.error("Error sharing:", error);
+      });
+    } else {
+      console.log("test");
```
Comment (rubek-joshi) [inline src/screens/GameOver/components/Share.tsx:45]: Remove console.log

File: src/screens/GameOver/index.tsx [hunk 7]
```diff
@@ -2,7 +2,8 @@ import { useGameContext } from "hooks";
 import styles from "./GameOver.module.scss";
```
Comment (rubek-joshi) [inline src/screens/GameOver/index.tsx:2]: Relative imports after other imports

**Review by rubek-joshi — CHANGES_REQUESTED** [review]

File: src/screens/GameOver/index.tsx [hunk 8]
```diff
@@ -18,45 +19,66 @@ export const GameOver = () => {
     };
   }, []);
 
+  const [share, setShare] = useState(false);
+
+  const toggleShare = () => {
+    setShare((prev) => !prev);
+  };
```
Comment (prakash-2021) [inline src/screens/GameOver/index.tsx:26]: not necessary but u can simply do 
`
const toggleShare = () => setShare((prev) => !prev);
`
decreased 2 line of code

File: src/screens/GameOver/index.tsx [hunk 9]
```diff
@@ -18,45 +19,66 @@ export const GameOver = () => {
...
+              <Button
+                color=""
+                size="fixed-width"
+                onClick={handlePlayAgain}
+                label="Try Again"
+              />
+            </div>
           </div>
         </div>
       </div>
```
Comment (prakash-2021) [inline src/screens/GameOver/index.tsx:84]: spacing

File: src/screens/Quiz/components/Score.module.scss [hunk 10]
```diff
@@ -1,10 +1,16 @@
...
 
 .container {
-  @apply flex items-center gap-3;
+  @apply flex items-center;
+  gap: main.em(12);
 }
 
 .label {
-  font-size: main.rem(22px);
+  font-size: main.em(22px);
```
Comment (prakash-2021) [inline src/screens/Quiz/components/Score.module.scss:9]: remove 22px to 22 to maintain consistency

File: src/styles/_spacing.scss [hunk 11]
```diff
@@ -1 +1,58 @@
...
+  }
+}
+
+.ca-mt {
+  &--56 {
+    margin-top: functions.em(56);
+  }
+}
+
+.ca-pt--64 {
```
Comment (prakash-2021) [inline src/styles/_spacing.scss:31]: use &-- here too
if new pt added will easier in future

**Review by prakash-2021 — CHANGES_REQUESTED** [review]

File: src/screens/GameOver/index.tsx (hunk 8 above)
Comment (rubek-joshi) [reply src/screens/GameOver/index.tsx:26]: Can further be improved with the use of `useBoolean` hook from usehook-ts library.

**Review by rubek-joshi — APPROVED** [review]

**Pull request merged**